### ISSUE TITLE
[P1][dogfood] C# extractor drops the receiver: every method/property has parent_class=None

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#766]
**Severity:** P1  |  **Surface:** C# extraction correctness
_Found by parallel dogfood audit round 2 on develop @ 471ac5f5; adversarially re-reproduced + dup-checked by an independent verifier._

### Observed
Prints {None}. All 19 extracted C# methods/properties have parent_class=None (Id, Name, Email, CreatedAt, PhoneNumber, DisplayName, User ctor, UpdateEmail, IsValid, IUserRepository.GetById, UserSettings, etc.). The C# parser DOES detect the owning types (classes=['User','IUserRepository','UserRole','UserSettings']) but never assigns any of them as the parent_class of their members, so members cannot be attributed to their container. By contrast the Ruby extractor on examples/Sample.rb with the same flags populates parent_class correctly ({User, AdminUser, Session, UserRepository}), confirming a C#-specific receiver drop.

### Expected
Each C# member should carry its owning type: User.UpdateEmail -> parent_class='User', IUserRepository.GetById -> parent_class='IUserRepository', the UserSettings constructor -> parent_class='UserSettings', etc. Dropping the receiver makes ClassName.method queries and structural attribution impossible for C# (CLAUDE.md root-cause class A).

### Repro
```bash
cd /Users/aisheng.yu/git-private/tsa-dogfood && uv run python -m tree_sitter_analyzer examples/Sample.cs --advanced --table json 2>/dev/null | python3 -c "import json,sys; d=json.load(sys.stdin); print(set(m.get('parent_class') for m in d['methods']))"
```

### Verifier notes
Re-ran the exact repro: it prints {None}. All 19 extracted C# methods/properties (Id, Name, Email, User ctor, UpdateEmail, IUserRepository.GetById, UserSettings, etc.) carry parent_class=None. The C# extractor DOES detect the containers (classes list = ['User','IUserRepository','UserRole','UserSettings']), so the owning type exists in the parse but is never attributed to its members — a genuine receiver drop, not a missing-class artifact. Differential confirmed: the Ruby extractor on examples/Sample.rb with the identical --advanced --table json flags correctly populates parent_class ({User, AdminUser, Session, UserRepository}), so this is C#-specific (CLAUDE.md root-cause class A: methods flattened, receiver lost). Not a flag/user error: --advanced --table json is the documented invocation and works for Ruby. Dedup: round-1 issues #734-#757 cover JS/Go/C/C++ extraction defects; the nearest analogs are Go-specific (#750 generic-receiver loss, #749 interface phantom methods) — none are C#, and no open issue mentions C# or parent_class. Distinct finding.

**Comment by aimasteracc** [comment]
Fixed by PR #823 — C# methods/properties now carry receiver_type from owning class/interface/struct/record.
